web/app/dash/submit/page.tsx

```diff
@@ -9,22 +9,24 @@ import { getDownloadURL, ref as sRef, uploadBytes } from "firebase/storage";
 import { useFirebase } from "@/app/firebase/context";
 import { useUserInfo } from "@/app/login/auth/context";
 
-async function uploadAlgo(
+async function uploadFile(
   database: any,
   storage: any,
   uid: string,
   file: File,
 ) {
-  const fileRef = push(ref(database, `users/${uid}/algos`));
+  const fileNameWithoutType = file.type.split("/")[0];
+  const fileRef = push(ref(database, `users/${uid}/${fileNameWithoutType}`));
   if (!fileRef) {
     return { downloadURL: "", fileIdKey: "", fileRef: "" };
   }
   const fileIdKey: string = `${uid}/${fileRef.key}` || ""; //bad practice
   const storageRef = sRef(storage);
   const fileType = file.type.split("/")[1];
-  const algoRef = sRef(storageRef, `algos/${fileIdKey}.${fileType}`);
+  const storageFileRef = sRef(storageRef, `${fileNameWithoutType}/${fileIdKey}.${fileType}`);
```

Comment:
This shouldn't be allowed, there could be conflicts if two files with the same filename are uploaded. You should be using fileIdKey as the key because it's unique